packages/components/src/internal/components/chart/ChartBuilderModal.tsx

```diff
@@ -0,0 +1,648 @@
+import React, { FC, Fragment, memo, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import classNames from 'classnames';
+
+import { PermissionTypes, Utils } from '@labkey/api';
+
+import { generateId } from '../../util/utils';
+import { LABKEY_VIS } from '../../constants';
+import { Modal } from '../../Modal';
+
+import { SelectInput, SelectInputOption } from '../forms/input/SelectInput';
+
+import { naturalSortByProperty } from '../../../public/sort';
+
+import { LoadingSpinner } from '../base/LoadingSpinner';
+
+import { deleteChart, saveChart } from '../../actions';
+
+import { QueryModel } from '../../../public/QueryModel/QueryModel';
+import { RequiresModelAndActions } from '../../../public/QueryModel/withQueryModels';
+
+import { useServerContext } from '../base/ServerContext';
+import { hasPermissions } from '../base/models/User';
+
+import { Alert } from '../base/Alert';
+import { FormButtons } from '../../FormButtons';
+
+import { getContainerFilter } from '../../query/api';
+
+import { SVGIcon } from '../base/SVGIcon';
+
+import { ChartConfig, ChartQueryConfig, GenericChartModel } from './models';
+
+interface AggregateFieldInfo {
+    name: string;
+    value: string;
+}
+
+export interface ChartFieldInfo {
+    aggregate?: AggregateFieldInfo;
+    // allowMultiple?: boolean;
```

Comment:
We should delete this or at least leave a comment that indicates why we're keeping a field commented out.

Reply:
comment added